src/sage/groups/perm_gps/partn_ref/data_structures.pxd

```diff
@@ -136,6 +136,37 @@ cdef inline void OP_join(OrbitPartition *OP, int m, int n) noexcept:
     if m_root != n_root:
         OP.num_cells -= 1
 
+
+cdef inline void OP_make_set(OrbitPartition *OP) noexcept:
+    cdef int i, n = OP.degree
+    cdef int *new_parent, *new_rank, *new_mcr, *new_size
+
+    cdef int *int_array = <int *> sig_malloc( 4*(n+1) * sizeof(int) )
+    if int_array is NULL:
```

Comment:
Do you really want to call free on NULL ?

Reply:
Ah, sorry, I'll fix that.